lib/galaxy/datatypes/data.py

```diff
@@ -1205,6 +1205,143 @@ def regex_line_dataprovider(
 class Directory(Data):
```

Comment:
Why does the `Directory` type does not define an `ext` attribute?

Do you know how the data is stored on disk? Is it zipped? Since one application of this is to hold large amounts of files it would actually be cool .. I guess. 

Where in the code is this managed?

Reply:
> Why does the Directory type does not define an ext attribute?

Oh, I guess it *should* define one. There just wasn't one initially and I didn't even think about it.



> Do you know how the data is stored on disk? Is it zipped? Since one application of this is to hold large amounts of files it would actually be cool .. I guess.
> 
> Where in the code is this managed?

The boring answer is: there's nothing of that type of magic going on here. The whole folder structure is simply stored at the extra_files_path. Some clever compression might be nice to have, yes, but I have no clue how you would implement this efficiently without knowing how often files get used, which depends on what tools you run on that data and what the user wants to do with it (viewing, downloading, etc.). So this sounds like something for a *really* clever datatype class, not like something I would implement.

Reply:
Compression is not galaxy's responsibility, let's not go down that route.

Reply:
added the file_ext attribute